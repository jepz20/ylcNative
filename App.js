import React from 'react'
import { StyleSheet, View, UIManager } from 'react-native'
import { Provider } from 'react-redux'

import store from './store'
import MainScreen from './screens/MainScreen'
import CalculatorScreen from './screens/CalculatorScreen'

export default class App extends React.Component {
  componentDidMount () {
    UIManager.setLayoutAnimationEnabledExperimental &&
      UIManager.setLayoutAnimationEnabledExperimental(true)
  }

  render () {
    return (
      <Provider store={store}>
        <View style={styles.container}>
          <MainScreen />
          <CalculatorScreen />
        </View>
      </Provider>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})
