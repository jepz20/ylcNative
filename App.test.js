import React from 'react'
import App from './App'
import { UIManager } from 'react-native'
import renderer from 'react-test-renderer'

it('renders without crashing when setLayoutAnimationEnabledExperimental doesnt exists', () => {
  const rendered = renderer.create(<App />).toJSON()
  expect(rendered).toMatchSnapshot()
})

it('renders without crashing when it doesnt exists UIManager', () => {
  UIManager.setLayoutAnimationEnabledExperimental = jest.fn()
  const rendered = renderer.create(<App />).toJSON()
  expect(rendered).toBeTruthy()
})
