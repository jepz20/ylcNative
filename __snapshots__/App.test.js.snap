// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when setLayoutAnimationEnabledExperimental doesnt exists 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 60,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "green",
          "flex": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          testID={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "#006400",
                "borderColor": "#005a00",
                "borderWidth": 1,
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "height": 35,
                  "justifyContent": "space-around",
                  "marginTop": 10,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 65,
                    "fontWeight": "bold",
                  }
                }
              >
                8000
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 40,
                  "justifyContent": "flex-end",
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Jose
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          testID={undefined}
        >
          <View
            style={
              Object {
                "backgroundColor": "#006400",
                "borderColor": "#005a00",
                "borderWidth": 1,
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "height": 35,
                  "justifyContent": "space-around",
                  "marginTop": 10,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 65,
                    "fontWeight": "bold",
                  }
                }
              >
                8000
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 40,
                  "justifyContent": "flex-end",
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "bold",
                  }
                }
              >
                Eduardo
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 5,
            "flexDirection": "row",
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={undefined}
            testID={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={undefined}
            testID={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                  "marginTop": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "color": "grey",
                    "fontSize": 30,
                  }
                }
              >
                .
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
